save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt

```diff
@@ -5,137 +5,407 @@
     /**
-     * Whether error modal from backend is visible
+     * Binary file of project
+     */
+    var binaryFile: File?
+
+    /**
+     * Property file for project
+     */
+    var propertyFile: File?
+
+    /**
+     * Message of error
+     */
+    var errorMessage: String
+
+    /**
+     * Flag to handle error
      */
     var isErrorOpen: Boolean
 
     /**
-     * Whether error text from backend is visible
+     * Error label
      */
-    var errorText: String
+    var errorLabel: String
+
+    /**
+     * Flag to handle loading
+     */
+    var isLoading: Boolean
 }
 
 /**
  * A functional RComponent for project view
+ * Each modal opening call re render full page, that why we need to use state for all fields
  *
  * @return a functional component
  */
 @OptIn(ExperimentalJsExport::class)
 @JsExport
-class ProjectView : RComponent<ProjectProps, ProjectViewState>() {
-    private var pathToPropertyFile: String? = null
+@Suppress("CUSTOM_GETTERS_SETTERS")
+class ProjectView : RComponent<ProjectExecutionRouteProps, ProjectViewState>() {
+    private val testTypesList = listOf("Class", "Comment", "Function", "Variable", "Enum", "Space", "Style", "Another")
+    private var pathToProperty: String? = null
+    private var gitUrlProject: String? = null
+    private val selectedTypes: MutableList<String> = mutableListOf()
+    private var gitDto: GitDto? = null
+
+    private var numberOpenningCard: Int = 1  // 1 - first card, 2 - second card, 3 - none card was opened
+    private var project = Project("stub", "stub", "stub", "stub")
     private lateinit var responseFromExecutionRequest: Response
 
     init {
         state.isErrorOpen = false
-        state.errorText = ""
+        state.errorMessage = ""
+        state.errorLabel = ""
+
+        state.isLoading = true
+    }
+
+    override fun componentDidMount() {
+        GlobalScope.launch {
+            val headers = Headers().also {
+                it.set("Accept", "application/json")
+                it.set("Content-Type", "application/json")
+            }
+            project = get("${window.location.origin}/getProject?name=${props.name}&owner=${props.owner}", headers)
+                .json()
+                .await()
+                .unsafeCast<Project>()
+            val jsonProject = JSON.stringify(project)
+            gitDto = post("${window.location.origin}/getGit", headers, jsonProject)
+                .json()
+                .await()
+                .unsafeCast<GitDto?>()
+            setState { isLoading = false }
+        }
     }
 
+    @Suppress("ComplexMethod", "TOO_LONG_FUNCTION")
     private fun submitExecutionRequest() {
-        val executionRequest = ExecutionRequest(
-            project = props.executionRequest.project,
-            gitRepository = props.executionRequest.gitRepository,
-            propertiesRelativePath = pathToPropertyFile ?: "save.properties"
-        )
+        if (numberOpenningCard == 0) {
```

Comment:
Is this situation still possible?

Reply:
Yes, `data-parent` allow to close both of cards